framework/validators/UniqueValidator.php

```diff
@@ -133,7 +145,24 @@ public function validateAttribute($model, $attribute)
         }
 
         if ($exists) {
-            $this->addError($model, $attribute, $this->message);
+            if (is_array($targetAttribute)) {
+                $attributeCombo = [];
```

Comment:
I can agree if you can come up with a good name for this separate method (`addError` cannot be used since its already defined in base class) and IMO this method should be made private so that potential refactorings of `UniqueValidator` class can have more freedom in splitting/merging class methods, etc., without worrying of breaking backward compatibility if someone used this `protected` method in their code.